src/backend/marsha/core/tests/test_command_send_reminders.py

```diff
@@ -0,0 +1,995 @@
+        # only key 1 has been added
+        self.assertEqual([REMINDER_2], registration.reminders)
+
+        # we call the command, no new email should be sent
+        call_command("send_reminders")
+        self.assertEqual(len(mail.outbox), 1)
+
+    @patch(
+        "marsha.core.management.commands.send_reminders.send_mail",
+        side_effect=smtplib.SMTPException("Error SMTPException"),
+    )
+    def test_send_reminders_send_email_fails(self, _mock_send_mail):
+        """send_mail fails, we make sure the error is raised and should_send_reminders is
+        disabled."""
+        video = VideoFactory(
+            live_state=IDLE,
+            live_type=RAW,
+            starting_at=timezone.now() + timedelta(days=2),
+        )
+
+        registration = LiveRegistrationFactory(
+            anonymous_id=uuid.uuid4(),
+            created_on=timezone.now() - timedelta(days=32),
+            email="sarah@test-fun-mooc.fr",
+            is_registered=True,
+            should_send_reminders=True,
+            video=video,
+        )
+
+        out = StringIO()
+        call_command("send_reminders", stdout=out)
+        self.assertEqual(len(mail.outbox), 0)
+        self.assertIn(
+            f"Sending email for liveregistration {registration.id} for video "
+            f"{registration.video.id} step {REMINDER_3}",
+            out.getvalue(),
+        )
+        self.assertIn("Mail failed sarah@test-fun-mooc.fr", out.getvalue())
+        registration.refresh_from_db()
+        # key has been added
+        self.assertEqual([REMINDER_3, REMINDER_ERROR], registration.reminders)
+        out.close()
+
+        # we call the command, no email should be sent
+        call_command("send_reminders")
+        self.assertEqual(len(mail.outbox), 0)
+
+    def test_send_reminders_simultanously(self):
+        """We simulate that query to update doesn't have any match results
+        and make sure no emails are sent"""
+        video = VideoFactory(
+            live_state=IDLE,
+            live_type=RAW,
+            starting_at=timezone.now() + timedelta(days=2),
+        )
+
+        registration = LiveRegistrationFactory(
+            anonymous_id=uuid.uuid4(),
+            created_on=timezone.now() - timedelta(days=32),
+            email="sarah@test-fun-mooc.fr",
+            is_registered=True,
+            should_send_reminders=True,
+            video=video,
+        )
+        patcher = patch(
+            "marsha.core.management.commands.send_reminders.Command.query_to_update_step",
+            return_value=0,
+        )
+        patcher.start()
+        out = StringIO()
+        call_command("send_reminders", stdout=out)
+        self.assertNotIn(
+            f"Sending email for liveregistration {registration.id} for video "
+            f"{registration.video.id} step {REMINDER_3}",
+            out.getvalue(),
+        )
+        self.assertEqual(len(mail.outbox), 0)
+
+        # stop mock
+        patcher.stop()
```

Comment:
```suggestion
        with patch(
            "marsha.core.management.commands.send_reminders.Command.query_to_update_step",
            return_value=0,
        ):
            out = StringIO()
            call_command("send_reminders", stdout=out)
           self.assertNotIn(
                f"Sending email for liveregistration {registration.id} for video "
                f"{registration.video.id} step {REMINDER_3}",
                out.getvalue(),
            )
            self.assertEqual(len(mail.outbox), 0)
```

It doesn't work like this ?

Reply:
yes it works I read the doc didn't know this one was preferred